### ISSUE TITLE
Add documentation for release-0.4

### ISSUE THREAD

**Issue by tabergma** [zalandoresearch/flair#285]
Add documentation for 

- [x] hyper parameter selection
- [x] how to resume training?
- [x] new language models
- [x] pretrained classification models
- [x] NLP data set downloader
- [ ] multi data set training
- [x] learning rate finder
- [x] custom optimizers
- [x] how to use BERT embeddings?
- [x] how to use ELMo embeddings?

**Comment by stefan-it** [comment]
One question: could you add a full example for the hyper parameter search (for `SequenceTaggerParamSelector`)? I don't really know where to pass the `StackedEmbeddings` :(

**Comment by tabergma** [comment]
@stefan-it Good point! You need to add the embeddings to the search space like this 
```
search_space.add(Parameter.EMBEDDINGS, hp.choice, options=[
    StackedEmbeddings[ WordEmbeddings('en') ], 
    StackedEmbeddings[ WordEmbeddings('en'), CharLMEmbeddings('news-forward'), CharLMEmbeddings('news-backward') ]
])
```
If you don't want to test different kind of embeddings, simply pass just one embedding option to the search space, which will than be used in every test run. Here is an example:
```python
search_space.add(Parameter.EMBEDDINGS, hp.choice, options=[
    StackedEmbeddings[ WordEmbeddings('en'), CharLMEmbeddings('news-forward'), CharLMEmbeddings('news-backward') ]
])
```
Does this help?

**Comment by stefan-it** [comment]
Thanks for that :+1: 

I'm currently training models for "Fine-grained POS Tagging of German Tweets" (see #51) with the `SequenceTaggerParamSelector` :)
